web/src/OverviewTable.tsx

```diff
@@ -551,13 +618,54 @@ function viewToRowValues(
   )
 }
 
-export default function OverviewTable(props: OverviewTableProps) {
-  let logStore = useLogStore()
-  const data = React.useMemo(() => viewToRowValues(props.view, logStore), [
-    props.view.uiResources,
-    props.view.uiButtons,
-  ])
+function getResourceLabels(resource: UIResource) {
+  const uiLabels = asUILabels({ labels: resource.metadata?.labels })
+  return getUILabels(uiLabels)
+}
```

Comment:
this function is kind of confusing!
1. the names `getResourceLabels`, `asUILabels`, and `getUILabels` don't really distinguish themselves from each other
2. `getUILabels(uiLabels)` - getting ui labels from ui labels doesn't make much sense!

Some options:
* At a glance, it seems pretty clear this is just getting labels from a resource so maybe just adding a `: []string` return type annotation would clear things up a bit
* The only other place that calls `asUILabels` also just passes its result into `getUILabels`, so maybe just `asUILabels`, `getUILabels`, and `getResourceLabels` into one function, to clean up the naming?
* rename `UILabels`/`asUILabels`, which doesn't seem like the right name anyway - at that point, it represents all labels, not just ones used by the UI

Reply:
you're right! i know the typescript gymnastics are a little goofy with both the type predicate and the casting function. i'll added better commenting and created a single helper function that's used instead of `as`/`isUILabels`.